src/coreclr/jit/fgbasic.cpp

```diff
@@ -810,36 +819,45 @@ class FgStack
     {
         return depth >= 1;
     }
+    void DuplicateTop()
+    {
+        FgSlot const top = Pop();
+        Push(top);
+        Push(top);
+    }
+    FgSlot Pop()
+    {
+        if (depth == 0)
+        {
+            // We don't maintain a 100% precise stack
```

Comment:
What's stopping us from doing so? We are resolving the call tokens and so should know argument counts and therefore how many values to push/pop for any given opcode, right?

Reply:
@tannergooding I hope eventually it will be so 🙂 maybe even some sort of a lightweight IR - at the moment it handle only a limitted set of opcodes and pushes just a single SLOT_UNKNOWN for everything else.